facade/src/main/java/org/jboss/pnc/facade/providers/ProductVersionProviderImpl.java

```diff
@@ -158,6 +164,23 @@ private void validateGroupConfigsBeforeUpdating(String id, ProductVersion restEn
         }
     }
 
+    private void validateMilestone(String id, ProductVersion entity) {
+        if (entity.getCurrentProductMilestone() != null) {
+            Integer newMilestoneId = mapper.getIdMapper().toEntity(entity.getCurrentProductMilestone().getId());
```

Comment:
`mapper` is not ProductMilestoneMapper, is it?

Reply:
depends on time when you look at it